src/test/java/teammates/sqlui/webapi/UnpublishFeedbackSessionActionTest.java

```diff
@@ -0,0 +1,170 @@
+
+    private Course course1;
+    private FeedbackSession feedbackSession1;
+    private FeedbackSession feedbackSession2;
+
+    @Override
+    protected String getActionUri() {
+        return Const.ResourceURIs.SESSION_PUBLISH;
+    }
+
+    @Override
+    protected String getRequestMethod() {
+        return DELETE;
+    }
+
+    @BeforeMethod
+    void setUp() throws InvalidParametersException, EntityDoesNotExistException {
+        course1 = generateCourse1();
+        feedbackSession1 = generateSession1InCourse(course1);
+        feedbackSession2 = generateSession2InCourse(course1);
+        Instructor instructor1 = generateInstructor1InCourse(course1);
+
+        when(mockLogic.getFeedbackSession(feedbackSession1.getName(), course1.getId())).thenReturn(feedbackSession1);
+        when(mockLogic.unpublishFeedbackSession(
+                feedbackSession1.getName(), course1.getId())).thenReturn(feedbackSession2);
+        when(mockLogic.getInstructorByGoogleId(
+                course1.getId(), instructor1.getAccount().getGoogleId())).thenReturn(instructor1);
+
+        loginAsInstructor(instructor1.getAccount().getGoogleId());
+    }
+
+    @Test
+    protected void testExecute() {
+        ______TS("Typical case");
+
+        String[] params = {
+                Const.ParamsNames.COURSE_ID, course1.getId(),
+                Const.ParamsNames.FEEDBACK_SESSION_NAME, feedbackSession1.getName(),
+        };
+
+        UnpublishFeedbackSessionAction unpublishFeedbackSessionAction = getAction(params);
+
+        JsonResult result = getJsonResult(unpublishFeedbackSessionAction);
+        FeedbackSessionData feedbackSessionData = (FeedbackSessionData) result.getOutput();
+
+        when(mockLogic.getFeedbackSession(feedbackSession1.getName(), course1.getId())).thenReturn(feedbackSession2);
+
+        assertEquals(feedbackSessionData.getFeedbackSessionName(), feedbackSession1.getName());
+        assertEquals(FeedbackSessionPublishStatus.NOT_PUBLISHED, feedbackSessionData.getPublishStatus());
+        assertFalse(mockLogic.getFeedbackSession(feedbackSession1.getName(), course1.getId()).isPublished());
+
+        ______TS("Typical case: Session is already unpublished");
+        // Attempt to publish the same session again.
+
+        result = getJsonResult(getAction(params));
+        feedbackSessionData = (FeedbackSessionData) result.getOutput();
+
+        assertEquals(feedbackSessionData.getFeedbackSessionName(), feedbackSession1.getName());
+        assertEquals(FeedbackSessionPublishStatus.NOT_PUBLISHED, feedbackSessionData.getPublishStatus());
+        assertFalse(mockLogic.getFeedbackSession(feedbackSession1.getName(), course1.getId()).isPublished());
+    }
+
+    @Test
+    public void testExecute_invalidRequests_shouldFail() {
+        ______TS("non existent session name");
+
+        String randomSessionName = "randomName";
+
+        assertNotNull(mockLogic.getFeedbackSession(feedbackSession1.getName(), course1.getId()));
```

Comment:
Same here, don't test mockito. The original tests were integration tests, if you want to migrate them they should go into the integration tests folder instead. Unit tests cannot be written in the same way as the old tests. Replacing logic with mockLogic does not make this a unit test.